File: src/cognition/layer/mood.py

def compute_snark(cpu, temp):
    snark = 0.0
    if cpu is not None:
        snark += (cpu / 100) * 3
    if temp is not None and temp >= 30:
        snark += min((temp - 30) / 10 * 2, 2)
    return min(snark, 5.0)

def compute_mood(snark):
    if snark < 1.0:
        return "calm"
    elif snark < 2.0:
        return "grumpy"
    elif snark < 3.0:
        return "irritated"
    elif snark < 4.0:
        return "snarky"
    elif snark < 4.7:
        return "hostile"
    else:
        return "furious"

File: src/bot/character/mood.js
export function computeSnark(cpu, temp) {
  let snark = 0.0;

  if (cpu !== null && cpu !== undefined) {
    snark += (cpu / 100) * 3;
  }

  if (temp !== null && temp !== undefined && temp >= 30) {
    snark += Math.min(((temp - 30) / 10) * 2, 2);
  }

  return Math.min(snark, 5.0);
}

export function computeMood(snark) {
  if (snark < 1.0) {
    return 'calm';
  } else if (snark < 2.0) {
    return 'grumpy';
  } else if (snark < 3.0) {
    return 'irritated';
  } else if (snark < 4.0) {
    return 'snarky';
  } else if (snark < 4.7) {
    return 'hostile';
  } else {
    return 'furious';
  }
}
